backend-lib/src/main/rust/lib.rs

```diff
@@ -642,6 +644,46 @@ pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_getCurren
     unwrap_exc_or(&mut env, res, ptr::null_mut())
 }
 
+/// Generates and returns an ephemeral address for one-time use, such as when receiving a swap from
+/// a decentralized exchange.
+#[unsafe(no_mangle)]
+pub unsafe extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_getSingleUseTaddr<
```

Comment:
We haven't declared these as `unsafe` in the past, because that only affects the callers of these methods which will only ever be the JNI side, and the `jni` crate's types are what insulate us from any unsafety inside the function. If we do want to start declaring these as `unsafe extern`s then clippy requires us to add `# Safety` sections. Either do that, or:
```suggestion
pub extern "C" fn Java_cash_z_ecc_android_sdk_internal_jni_RustBackend_getSingleUseTaddr<
```
(and similarly below).